main/context_processors.py

```diff
@@ -0,0 +1,21 @@
+from daycare.models import DayCare
+from dogowner.models import DogOwner
+
+
+def navbar_extras(request):
+    navbar_picture_url = navbar_name = None
+    if request.user.is_authenticated:
+        if DogOwner.objects.filter(user=request.user).exists():
+            dog_owner = DogOwner.objects.filter(user=request.user).first()
+            navbar_name = dog_owner.__str__()
+            navbar_picture_url = dog_owner.get_dog_owner_profile_image_url()
+        elif DayCare.objects.filter(user=request.user).exists():
```

Comment:
maybe you can just use else here (if a user is logged and he is not a dogowner we know he is a daycare), and avoid line 6 of init variables to None